packages/odyssey-react-mui/src/labs/SideNav/SideNav.tsx

```diff
@@ -402,31 +407,90 @@ const SideNav = ({
     [isCompact],
   );
 
-  const processedSideNavItems = useMemo(
-    () =>
-      sideNavItems.map((item) => ({
-        ...item,
-        children: item.children?.map((childProps) => {
-          return (
+  const setSelectedInChildItems = useCallback(
+    ({
+      nestedNavItems,
+      selectedItemId,
+    }: {
+      nestedNavItems: Array<Omit<SideNavItem, "startIcon" | "endIcon">>;
+      selectedItemId: string;
+    }) => {
+      const childItems = nestedNavItems.map((childItem) => {
+        if (childItem.isSelected) {
+          childItem.isSelected = false;
+        }
+        if (childItem.id === selectedItemId) {
+          childItem.isSelected = true;
+        }
+        if (childItem.nestedNavItems) {
+          setSelectedInChildItems({
+            nestedNavItems: childItem.nestedNavItems,
+            selectedItemId,
+          });
+        }
+        return childItem;
+      });
+      return childItems;
+    },
+    [],
+  );
+
+  const setSelectedItem = useCallback(
+    (selectedItemId: string) => {
+      const newNavItems = sideNavItemsList.map((item) => {
+        if (item.isSelected) {
+          item.isSelected = false;
+        }
+        if (item.id === selectedItemId) {
+          item.isSelected = true;
+        }
+        if (item.nestedNavItems) {
+          item.nestedNavItems = setSelectedInChildItems({
+            nestedNavItems: item.nestedNavItems,
+            selectedItemId,
+          });
+        }
+        return item;
+      });
+      updateSideNavItemsList(newNavItems);
+    },
+    [sideNavItemsList, setSelectedInChildItems],
+  );
```

Comment:
Is there a reason we're not using `setSelectedInChildItems` here?

I'd like to avoid duplicate logic:
```suggestion
  const setSelectedItem = useCallback(
    (selectedItemId: string) => {
      updateSideNavItemsList(setSelectedItem(sideNavItemsList));
    },
    [sideNavItemsList, setSelectedInChildItems],
  );
```

Reply:
The main SideNavItem and the child nested NavItems are of different types. We get into type errors when trying to call this recursively.